pkg/invariantlibrary/allowedbackenddisruption/matches_test.go

```diff
@@ -274,16 +275,20 @@ func TestDisruptionDataFileParsing(t *testing.T) {
 			foundMetalOVN = true
 		}
 	}
-
+	currentRelease = currentReleaseFromMap(releasesInQueryResults)
 	assert.Greater(t, dataOver100Runs, 5,
 		"expected at least 5 entries in query_results.json to have over 100 runs")
 	assert.True(t, foundAWSOVN, "no aws ovn job data in query_results.json")
 	assert.True(t, foundGCPOVN, "no gcp ovn job data in query_results.json")
 	assert.True(t, foundAzureOVN, "no azure ovn job data in query_results.json")
 	assert.True(t, foundMetalOVN, "no metal ovn job data in query_results.json")
-	assert.Equal(t, 1, len(releasesInQueryResults),
-		"expected only one Release in query_results.json")
-
+	if currentRelease <= "4.14" {
```

Comment:
Shouldn't really do comparisons like this on string representations of versions, 4.100 <= 4.14 for example, but in your case, I think != will suffice. If it's not 4.14 specifically, we should be good, as this would not run on any past releases.

Reply:
That said, I think this will immediately start failing when we switch to 5.15 if we haven't updated the data file, which could take up to a week.  That would suck.

Let's just assert <= 2 all the time and be done with it. We'll need to monitor the file ourselves. In future, put a TODO that this can be made to be Equals(2) for 4.15 and onward. Hopefully someone will see it someday.

Reply:
Just saw Forrest's comment too late. 

There should be a utility out there somewhere for comparing version strings, if you want to keep this check. 

For 4.14, it should be 1, we could test that. For 4.15, it should be 2, but only after the first week or so. The safest thing to do, if you want to keep the test specific, is to enforce 1 for that as well, it will pass when we branch, and then the first time the job submits a PR to origin, we'll see it fail, and someone will have to intervene to update the disruption data.

Reply:
For 4.14 we will actually get 4.13 data, just for non upgrades only since the query contains ` AND (MasterNodesUpdated != "N" OR FromRelease = "")` so the null MasterNodesUpdated records get dropped, except those with no FromRelease.

I'm wondering if we shouldn't manually update the file (run the ci-tools command locally and copy it over) the first time when we are ready and update the test at that time.